executor/sort.go

```diff
@@ -174,9 +182,31 @@ func (e *SortExec) prepareExternalSorting() (err error) {
 	e.buildKeyColumns()
 	e.rowPtrsInDisk = e.initPointersForListInDisk(e.rowChunksInDisk)
 	// partition sort
-	// Now only have one partition.
-	// The partition will be adjusted in the next pr.
-	err = e.readPartition(e.rowChunksInDisk, e.rowPtrsInDisk)
+	partStartPtr := 0
+	partStartOffset := e.rowChunksInDisk.GetOffsetOfRow(e.rowPtrsInDisk[partStartPtr])
+	for i := 0; i < len(e.rowPtrsInDisk); i++ {
+		size := e.rowChunksInDisk.GetOffsetOfRow(e.rowPtrsInDisk[i]) - partStartOffset
```

Comment:
Now we need to put all rows into disk, then read partition and sort it one by one, then merge sort.
There's a better adaptive procedure.

1. Read as mush as rows into memory.
2. If memory quota is triggered, sort these rows in memory and put them into disk as partition 1, then return to step 1
3. If memory quota is not triggered and child is consumed, sort these rows in memory as parition N, then do merge sort.

If there is only one partition in step 3, it works just like in-memory sort before.